packages/filebrowser/src/listing.ts

```diff
@@ -1123,24 +1123,59 @@ export class DirListing extends Widget {
     if (!files || files.length === 0) {
       return;
     }
-    const length = event.dataTransfer?.items.length;
-    if (!length) {
-      return;
-    }
-    for (let i = 0; i < length; i++) {
-      let entry = event.dataTransfer?.items[i].webkitGetAsEntry();
-      if (entry.isDirectory) {
-        console.log('currently not supporting drag + drop for folders');
-        void showDialog({
-          title: this._trans.__('Error Uploading Folder'),
-          body: this._trans.__(
-            'Drag and Drop is currently not supported for folders'
-          ),
-          buttons: [Dialog.cancelButton({ label: this._trans.__('Close') })]
-        });
+    event.preventDefault();
+
+    if (event.dataTransfer && event.dataTransfer.items) {
+      let items = event.dataTransfer.items;
+
+      const addDirectory = (item: any) => {
+        if (item.isDirectory) {
+          this._model.manager
+            .newUntitled({
+              path: this._model.path,
+              type: 'directory'
+            })
+            .then(async model => {
+              await this._manager.rename(
+                `${this._model.path}/Untitled Folder`,
```

Comment:
The promise gives you `model` which according to the [documentation of `newUntitled`](https://jupyterlab.readthedocs.io/en/latest/api/interfaces/docmanager.idocumentmanager.html#newuntitled) should return an [`IModel`](https://jupyterlab.readthedocs.io/en/latest/api/interfaces/services.contents.imodel.html) which has `.name` and `.path` attributes - use one of those instead of hard-coding the "Untitled Folder" name as otherwise it will not work when "Untitled folder" already exists and the previous action actually created "Untitled Folder 2".

As for rename, I am not sure if this expects relative path or not but this might be why it is not working.

Reply:
Okay, thanks for helping. I'll try and get back to you. 🙂

Reply:
I tried making the changes suggested by you by using `model.name` as well as `model.path` for renaming a directory but nothing worked.